src/sage/schemes/curves/projective_curve.py

```diff
@@ -437,12 +437,12 @@ def projection(self, P=None, PS=None):
                     if self.defining_polynomials()[i] != 0:
                         F = self.defining_polynomials()[i]
                 # find a point on which it doesn't vanish
-                l = list(PP.gens())
+                ll = list(PP.gens())
                 for i in range(n + 1):
-                    l[i] = 0
-                    while F(l) == 0:
-                        l[i] += 1
-                Q = PP(l)  # will be a point not on the curve
+                    ll[i] = 0
+                    while F(ll) == 0:
+                        ll[i] += 1
+                Q = PP(ll)  # will be a point not on the curve
```

Comment:
I (still) don't see any benefit in changes like this. There is no danger of confusion between `l` and `I`, the change doesn't actually alleviate the confusion from the same font, it could create needless merge conflicts, and now the variable makes less sense. In this case, a change to `L` would be more practical, but I (still) recommend just reverting this.